Enigmatry.Entry.CodeGeneration/Templates/Angular/Material/Angular.Form.Component.cshtml

```diff
@@ -171,40 +171,49 @@ import { map, throttleTime } from 'rxjs/operators';
         if (select.Options.HasDynamicValues)
         {
             @:private @(select.PropertyName)Options$ = new BehaviorSubject<any[]>([]);
-                @:@@Input()
-                @:get @(select.PropertyName)Options(): any[] { return this.@(select.PropertyName)Options$.value; }
-                @:set @(select.PropertyName)Options(value: any[]) { this.@(select.PropertyName)Options$.next(value); }
-            }
-            else
-            {
-                @:@@Input() @(select.PropertyName)Options: any[] = @Html.JsArray(select.Options.FixedOptions, option => $"{{ value: {option.Value ?? "null"}, displayName: {(@Html.Localize(option.DisplayName, Options.EnableI18N))} }}");
-            }
+            @:@@Input()
+            @:get @(select.PropertyName)Options(): any[] { return this.@(select.PropertyName)Options$.value; }
+            @:set @(select.PropertyName)Options(value: any[]) { this.@(select.PropertyName)Options$.next(value); }
+        }
+        else
+        {
+            @:@@Input() @(select.PropertyName)Options: any[] = @Html.JsArray(select.Options.FixedOptions, option => CreateSelectOptionTemplate(option));
+        }
 
-            if (select.Options.HasCustomValueAndDisplayKeys)
-            {
-                @:@@Input() @(select.PropertyName)OptionsConfiguration: SelectConfiguration = { valueProperty: '@(select.Options.OptionValueKey)', labelProperty: '@(select.Options.OptionDisplayKey)', sortProperty: '@(select.Options.OptionSortKey)' };
-            }
-            else
-            {
-                @:@@Input() @(select.PropertyName)OptionsConfiguration: SelectConfiguration = @(@Html.Raw(select.Options.DefaultOptionsAsString));
-            }
+        if (select.Options.HasCustomValueAndDisplayKeys)
+        {
+            @:@@Input() @(select.PropertyName)OptionsConfiguration: SelectConfiguration = { valueProperty: '@(select.Options.OptionValueKey)', labelProperty: '@(select.Options.OptionDisplayKey)', sortProperty: '@(select.Options.OptionSortKey)' };
         }
+        else
+        {
+            @:@@Input() @(select.PropertyName)OptionsConfiguration: SelectConfiguration = @(@Html.Raw(select.Options.DefaultOptionsAsString));
+        }
+    }
 
-        void CreateWrappers(FormControl control)
+    string CreateSelectOptionTemplate(SelectOption selectOption)
+    {
+        if (selectOption.Value != null && selectOption.Value.GetType() == typeof(string))
```

Comment:
I'm not sure about using type check for string only. Reason be, string interpolation will call ```ToString()``` on ```selectOption.Value``` because value is ```object``` Type. We always get a string in the interpolation expression.

The question is when do we need to display value without the single quotes ''. Maybe only for numbers (numeric types) and nulls.
 We can check IsNumericType() instead of checking on string. Something like this https://stackoverflow.com/questions/1749966/c-sharp-how-to-determine-whether-a-type-is-a-number